### PR TITLE
Loopback4 exercise7

### PR DESCRIPTION
**Pull request by iammukeshsahu**
Create a winston logger component. Use it in sequence.

Also create a middleware, that reads a cookie from headers, decrypt that cookie to get a userId, and then create a JWT from that cookie, and add it to the headers of the request as a Bearer token.

### CODE REVIEW

File: loopback4/src/sequence.ts [hunk 1]
```diff
@@ -32,32 +34,61 @@ export class MySequence implements SequenceHandler {
...
+    const [key, value] = pair.trim().split('=');
+    if (key === 'userId') {
+      return value;
     }
   }
+  return '';
+}
+
+function createJwtToken(userId: string): string {
+  const SECRET_KEY = 'Enter your secret key';
```
Comment (mlakdawala-sf) [inline loopback4/src/sequence.ts]: secret key should be env driven and should be some random values

Comment (mlakdawala-sf) [reply loopback4/src/sequence.ts]: same goes with token expiry

File: loopback4/src/sequence.ts [hunk 2]
```diff
@@ -32,32 +34,61 @@ export class MySequence implements SequenceHandler {
...
-      console.log('Error Time:', errorTime);
+      logger.error('Error Time:', { errorTime });
+      logger.error('Failed to load the application', { error })
+      this.reject(context, error);
+    }
+  }
+}
 
-      throw error;
+function decryptCookie(cookie: string): string {
```
Comment (mlakdawala-sf) [inline loopback4/src/sequence.ts]: Move functions to a separate class and can be injected through Depenendency injection of loopback

File: loopback4/src/sequence.ts [hunk 3]
```diff
@@ -32,32 +34,61 @@ export class MySequence implements SequenceHandler {
...
     try {
-      const allowedOrigin = process.env.ALLOWED_ORIGIN;
-      if (referer !== allowedOrigin) {
-        throw new Error('Referer not allowed');
+      logger.info('Request received:', {
+        method: context.request.method,
+        url: context.request.url,
+      });
+      const { cookie } = headers;
+      if (cookie) {
```
Comment (mlakdawala-sf) [inline loopback4/src/sequence.ts:51]: we should avoid cookies and use authorization header to pass the jwt, if mentioned in the curriculumn its fine then, else we should not follow this approach

**Review by mlakdawala-sf — CHANGES_REQUESTED** [review]

File: loopback4/src/sequence.ts (hunk 3 above)
Comment (iammukeshsahu) [reply loopback4/src/sequence.ts:51]: It is mentioned in curriculumn. you can check below text.
`Create a winston logger component. Use it in sequence.

Also create a middleware, that reads a cookie from headers, decrypt that cookie to get a userId, and then create a JWT from that cookie, and add it to the headers of the request as a Bearer token.`

**Review by mlakdawala-sf — APPROVED** [review]

**Pull request merged**